### PR TITLE
Add content improvements to error messages

### PR DESCRIPTION
**Pull request by Renny4Real**
## Changes in this PR:
This PR introduces a lot of changes in error content across the forms used to create a vacancy i.e.
- Job Specification Form
- Candidate Specification Form
- Application Details Form
- Review Job page
- Copy Job Form

It also introduces changes that:
- Remove unused translations.
- Refactor the translation file to reduce duplication and use a new format.
- Modify any error messages that were not noticed before.

The list of errors and their desired changes can be found [here](https://docs.google.com/spreadsheets/d/1-MeFAToarUINg-8Sif4K0ZTPudowE6FU_mf3Hhf_iOk/edit).

#### Before
Screenshot 2019-11-11 at 12 10 52

Screenshot 2019-11-11 at 12 11 12

### After
Screenshot 2019-11-11 at 12 11 29

Screenshot 2019-11-11 at 12 11 34

## Next steps:

- [ ] Check changes on Staging

**Any feedback/suggestions would be appreciated**

### CODE REVIEW

File: config/locales/activerecord.yml [hunk 1]
```diff
@@ -1,20 +1,68 @@
 en:
+  number:
+    format:
+      delimiter: ","
```
Comment (C-gyorfi) [inline config/locales/activerecord.yml]: Did this solve the number formatting? 🤔

Comment (Renny4Real) [reply config/locales/activerecord.yml]: No, it doesn't. 😞

Comment (Renny4Real) [reply config/locales/activerecord.yml]: Will remove this.

Comment (C-gyorfi) [reply config/locales/activerecord.yml]: In that case, I think it's better to remove, thanks 😄

**Pull request merged**